### PR TITLE
chore(local-dev): setup docker profiles and mock services

### PR DESCRIPTION
**Pull request by bryn-pins**
## Description of change

**Profiles:**
Setup 3 docker profiles and a make script to run them
- all
- appeals (forms web app and dependencies)
- comment  (web comment and dependencies, guessed with these having not worked on it)

**Service Mocks**
Mock clam av and pdf service to have slimmer option for running locally
Use these mocks with docker-compose.slim.yml

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,10 +19,26 @@ run:
...
+# only services required for web-comment
+web-comment:
+	docker compose --profile comment -f docker-compose.yml -f docker-compose.slim.yml up
+.PHONY: web-comment
+
+# only services required for appeals/forms-web-app
+appeals:
+	docker compose --profile appeals -f docker-compose.yml -f docker-compose.slim.yml up
+.PHONY: appeals
+
```
Comment (beejjacobs) [inline Makefile:41]: Should we make these npm scripts instead/as well?

File: Makefile [hunk 2]
```diff
@@ -19,10 +19,26 @@ run:
 		${CMD}
 .PHONY: run
 
+# all services
 serve:
-	docker-compose up
+	docker compose --profile all -f docker-compose.yml -f docker-compose.full.yml up
```
Comment (beejjacobs) [inline Makefile:24]: Does the order matter? do we need slim/full first, then the main one?

**Review by beejjacobs — COMMENTED** [review]
I like it! Couple of questions.

File: Makefile (hunk 1 above)
Comment (bryn-pins) [reply Makefile:41]: have added and documented in readme

File: Makefile (hunk 2 above)
Comment (bryn-pins) [reply Makefile:24]: my understanding is they're combined into one file when ran so order does matter due to overrides but I think this is the correct way round and currently doesn't matter?
https://docs.docker.com/engine/reference/commandline/compose/#specifying-multiple-compose-files

Comment (beejjacobs) [reply Makefile:24]: I was thinking more with the order things start in, but if this works that's all good.

**Review by beejjacobs — APPROVED** [review]
Looks good!

**Pull request merged**